File: backend/spring-boot/src/main/java/org/bugzkit/api/auth/jwt/redis/model/RefreshTokenStore.java
package org.bugzkit.api.auth.jwt.redis.model;

import java.io.Serial;
import java.io.Serializable;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.redis.core.RedisHash;
import org.springframework.data.redis.core.TimeToLive;
import org.springframework.data.redis.core.index.Indexed;

@Getter
@NoArgsConstructor
@AllArgsConstructor
@RedisHash(value = "RefreshTokenStore")
public class RefreshTokenStore implements Serializable {
  @Serial private static final long serialVersionUID = -1997218842142407911L;

  @Id private String refreshToken;

  @Indexed private Long userId;

  @Indexed private String ipAddress;

  @TimeToLive private long timeToLive;
}

File: backend/spring-boot-kotlin/src/main/kotlin/org/bootstrapbugz/api/auth/jwt/redis/model/RefreshTokenStore.kt
package org.bootstrapbugz.api.auth.jwt.redis.model

import org.springframework.data.annotation.Id
import org.springframework.data.redis.core.RedisHash
import org.springframework.data.redis.core.TimeToLive
import org.springframework.data.redis.core.index.Indexed
import java.io.Serial
import java.io.Serializable

@RedisHash(value = "RefreshTokenStore")
class RefreshTokenStore(
    @Id val refreshToken: String? = null,
    @Indexed val userId: Long? = null,
    @Indexed val ipAddress: String? = null,
    @TimeToLive val timeToLive: Long = 0
) : Serializable {

    companion object {
        @Serial
        private const val serialVersionUID = -1997218842142407911L
    }
}
